atom/browser/browser_linux.cc

```diff
@@ -33,18 +84,46 @@ void Browser::ClearRecentDocuments() {
 void Browser::SetAppUserModelID(const base::string16& name) {
 }
 
-bool Browser::RemoveAsDefaultProtocolClient(const std::string& protocol,
-                                            mate::Arguments* args) {
-  return false;
-}
-
 bool Browser::SetAsDefaultProtocolClient(const std::string& protocol,
                                          mate::Arguments* args) {
-  return false;
+  return SetDefaultWebClient(protocol);
 }
 
 bool Browser::IsDefaultProtocolClient(const std::string& protocol,
-                                      mate::Arguments* args) {
+                                        mate::Arguments* args) {
+  #if defined(OS_CHROMEOS)
+    return UNKNOWN_DEFAULT;
+  #else
+    base::ThreadRestrictions::AssertIOAllowed();
+
+    std::unique_ptr<base::Environment> env(base::Environment::Create());
+
+    std::vector<std::string> argv;
+    argv.push_back(kXdgSettings);
+    argv.push_back("check");
+    if (!protocol.empty()) {
+      argv.push_back(kXdgSettingsDefaultSchemeHandler);
+      argv.push_back(protocol);
+    }
+    argv.push_back(libgtkui::GetDesktopName(env.get()));
+
+    std::string reply;
+    int success_code;
+    bool ran_ok = base::GetAppOutputWithExitCode(base::CommandLine(argv),
+    &reply, &success_code);
+
+    if (!ran_ok || success_code != EXIT_SUCCESS) return false;
+
+    // Allow any reply that starts with "yes".
+    return base::StartsWith(reply, "yes", base::CompareCase::SENSITIVE)
+               ? true
+               : false;
+  #endif
+}
+
+// Todo implement
```

Comment:
🤔

Reply:
i'm not currently sure if it's possible to remove using `xdg-settings` sadly...docs don't reference anything about it and chromium doesn't implement any form of protocol removal